src/var.cc

```diff
@@ -85,6 +85,33 @@ void Var::Used(Evaluator* ev, const Symbol& sym) const {
   }
 }
 
+void Var::SetVisibilityPrefix(const std::vector<std::string>& prefixes, const char* name) {
+  std::vector<std::string> current_prefixes = VisibilityPrefix();
+  if (current_prefixes.size() == 0) {
+    visibility_prefix_ = prefixes;
+  } else if (current_prefixes != prefixes) {
+    ERROR("Visibility prefix conflict on variable: %s", name);
+  }
+}
+
+void Var::CheckCurrentReferencingFile(Loc loc, const char* name) {
+  std::vector<std::string> prefixes = VisibilityPrefix();
+  if (prefixes.size() == 0) {
+    return;
+  }
+  std::string referencingFile(loc.filename);
+  bool valid = false;
+  for (const std::string& prefix : prefixes) {
+    if (referencingFile.find(prefix) == 0) {
```

Comment:
It would seem that the preferred syntax here is: `if HasPrefix(referencingFile, prefix) {`

Reply:
Also from our discussion, this should only check path prefixes instead of string prefixes (i.e. ven is not a prefix of vendor/)

This is how visibility works in soong/bazel, see https://cs.android.com/android/platform/superproject/main/+/main:build/soong/android/visibility.go;l=174;drc=9a24d909369ae20566c85dfbfe8b14054ed6dd5d

Reply:
re: prefix matching -- sounds good to me.